src/rt-app.c

```diff
@@ -413,6 +433,83 @@ static int run_event(event_data_t *event, int dry_run,
 			pthread_yield();
 		}
 		break;
+	case rtapp_fork:
+		{
+			log_debug("fork");
+
+			/*
+			 * Check if the current thread reached its limit of
+			 * number of allowable forks.
+			 * We enforce a limit to prevent infinite loops.
+			 */
+			if (rdata->res.fork.nforks >= opts.forks_limit &&
+			    opts.forks_limit > -1) {
+				log_error("%s reached its fork limit (%d)", rdata->res.fork.tdata->name, opts.forks_limit);
+				exit(EXIT_FAILURE);
+			}
+
+			/*
+			 * If multiple threads race to fork, we must ensure
+			 * each one sees a unique index. Hence the lock.
+			 */
+			pthread_mutex_lock(&fork_mutex);
+			int new_thread_index = nthreads++;
+			threads = reallocarray(threads, nthreads, sizeof(*threads));
```

Comment:
Just discovered (by compiling on Debian 9.5.0) that reallocarray has been introduced in [Glibc 2.26](https://abi-laboratory.pro/?view=changelog&l=glibc&v=2.26). Debian ships with 2.24 and can't find this function.
Guess we should use realloc instead?

Reply:
Ah I was looking for realloc() in manpages when I saw it and I didn't realize it's new. Changed it and will update the branch shortly.

Thanks.